labgrid/resource/udev.py

```diff
@@ -407,6 +407,50 @@ def path(self):
 
         return None
 
+@target_factory.reg_resource
+@attr.s(eq=False)
+class USBSDWireDevice(USBResource):
+    """The USBSDWireDevice describes an attached SDWire device,
+    it is identified via USB using udev
+    """
+
+    disk_path = attr.ib(default=None)
+    control_serial = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        self.match['ID_VENDOR_ID'] = '04e8'
+        self.match['ID_MODEL_ID'] = '6001'
+        self.match['@ID_VENDOR_ID'] = '0424'
+        self.match['@ID_MODEL_ID'] = '2640'
+        super().__attrs_post_init__()
+
+    # Overwrite the avail attribute with our internal property
+    @property
+    def avail(self):
+        return bool(self.disk_path and self.control_serial)
+
+    # Forbid the USBResource super class to set the avail property
+    @avail.setter
+    def avail(self, prop):
+        pass
+
+    # Overwrite the poll function. Only mark the SDWire as available if both
+    # paths are available.
```

Comment:
@Emantor Does this fit here? I know it is copied from USBSDWireDevice. Maybe I misunderstand something, but I think it does not fit there either.

Reply:
I think it applies here too. We don't want to support partial availability for these devices, since we'll get errors in the USBMassStorageDriver otherwise.

Reply:
The comment itself is correct, I am unsure about the position though, it applies to the `avail` property above, doesn't it?

Reply:
It does, but was copied from USBSDMux. I am fine with merging this and fixing this later.